main.js

```diff
@@ -3371,21 +3371,60 @@ function startInstance(id, wakeUp) {
                                 (mode !== 'extension' || !procs[id].config.native.webInstance) &&
                                 mode !== 'once'
                             ) {
-                                logger.info(hostLogPrefix + ' Restart adapter ' + id + ' because enabled');
+                                if (code === EXIT_CODES.UNCAUGHT_EXCEPTION) {
+                                    // if its an uncaught exception, detect restart loop
+                                    procs[id].crashCount = procs[id].crashCount || 0;
+                                    procs[id].crashCount++;
+                                    logger.debug(`${hostLogPrefix} Crash count of ${id}: ${procs[id].crashCount}`);
+
+                                    if (procs[id].crashResetTimer) {
+                                        logger.debug(`${hostLogPrefix} Reset crash timer of ${id}, to be initialized anew`);
+                                        clearTimeout(procs[id].crashResetTimer);
+                                    }
+
+                                    // after 10 minutes without crash, we reset counter
+                                    logger.debug(`${hostLogPrefix} Initialize crash timer of ${id}`);
+                                    procs[id].crashResetTimer = setTimeout(() => {
+                                        logger.debug(`${hostLogPrefix} Cleared crash counter of ${id}, because 10 minutes no crash`);
+                                        procs[id].crashCount = 0;
```

Comment:
Better check here that procs[id] still exists ... there can be caes wherethe instance was moved to an other host in the meantime